pkg/xmtp/listener.go

```diff
@@ -185,12 +186,17 @@ func (l *Listener) processEnvelope(env *v1.Envelope) error {
 }
 
 func (l *Listener) shouldDeliver(messageContext interfaces.MessageContext, subscription interfaces.Subscription) bool {
-	if messageContext.ShouldPush != nil {
-		return *messageContext.ShouldPush
-	}
```

Comment:
Switched the ordering of the checks so that it first checks if it's the sender and if it is the sender it will return false immediately without checking anything else.